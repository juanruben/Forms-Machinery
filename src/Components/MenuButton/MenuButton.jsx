import React from 'react';
import PropTypes from 'prop-types';
import './MenuButton.scss';

function MenuButton(props) {
    const { onClick } = props;
    return (
        <button onClick={onClick} type="button" className="menu-icon">
            <i className="fas fa-bars" />
        </button>
    );
}

MenuButton.propTypes = {
    onClick: PropTypes.func.isRequired,
};

export default MenuButton;
