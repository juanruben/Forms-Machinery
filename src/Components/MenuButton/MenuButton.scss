@import "../../Assets/css/general.scss";

.menu-icon {
  background-color: $icafal-background;
  border: none;
  color: $icafal-orange;
  font-size: 30px;
  padding: 0;

  &:focus {
    outline: 0;
  }
}
