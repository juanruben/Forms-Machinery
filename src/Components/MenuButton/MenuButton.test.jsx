import React from 'react';
import { shallow } from 'enzyme';
import MenuButton from './MenuButton';

describe('MenuButton', () => {
    let wrapper;
    const onClick = jest.fn();

    it('should render ok', () => {
        wrapper = shallow(<MenuButton onClick={onClick} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('should call callback', () => {
        wrapper = shallow(<MenuButton onClick={onClick} />);
        wrapper.simulate('click');
        expect(onClick).toBeCalledTimes(1);
    });
});
