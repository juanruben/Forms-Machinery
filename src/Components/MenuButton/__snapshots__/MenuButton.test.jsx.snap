// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton should render ok 1`] = `
"<button onClick={[Function: mockConstructor]} type=\\"button\\" className=\\"menu-icon\\">
  <i className=\\"fas fa-bars\\" />
</button>"
`;
